contracts/Removal.sol

```diff
@@ -483,7 +580,10 @@ contract Removal is
     uint256 amount,
     bytes memory data
   ) public override whenNotPaused {
-    if (_msgSender() != address(_market)) {
+    if (
+      _msgSender() != address(_market) &&
+      hasRole({account: from, role: CONSIGNOR_ROLE})
```

Comment:
is this logic right? this `&&` will be false if the sender (`from`) does NOT have the `CONSIGNOR_ROLE`, which would bypass the revert and succeed if some random sender tried to use this function

Reply:
Oh, I don't think I even meant to change this function. Will revert the changes.